src/util/map.js

```diff
@@ -146,3 +146,20 @@ export function merge<K, V>(
   }
   return result;
 }
+
+/**
+ * Push a value into a map which has an array for every key.
```

Comment:
> Push a value into a map which has an array for every key.

This doesn’t make sense to me. “Has an array for every key” suggests
that the keys are arrays, and the use of “every” suggests that the map
is total over some (natural) domain, which certainly isn’t the case.

It sounds like you want to say that the _values_ of the map should all
be arrays, though this is already implied by the type. (I see how the
original text could be interpreted to mean this, but it wasn’t clear to
me, which suggests that it should be rewritten.)

What do you think of

```js
/**
 * Given a map whose values are arrays, push an element onto the array
 * corresponding to the given key. If the key is not in the map, first
 * insert it with value a new empty array.
 *
 * If the key is already in the map, its value will be mutated, not
 * replaced.
 */
```

?